.circleci/config.yml

```diff
@@ -1,14 +1,119 @@
-# This config is equivalent to both the '.circleci/extended/orb-free.yml' and the base '.circleci/config.yml'
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
 version: 2.1
 
 orbs:
-  node: circleci/node@4.1
+  # The Node.js orb contains a set of prepackaged CircleCI configuration you can utilize
+  # Orbs reduce the amount of configuration required for common tasks.
+  # See the orb documentation here: https://circleci.com/developer/orbs/orb/circleci/node
+  node: circleci/node@4.7.0
+
+jobs:
+  # Below is the definition of your job to build and test your app, you can rename and customize it as you want.
+  build-and-test:
+    docker:    
+      - image: cimg/node:15.1
+    steps:
+      # Checkout the code as the first step.
+      - checkout
+      # Build while on the deploy branch. Note - this will only be checked in if the deploy-branch-update runs.
+      - run:
+          name: Configure git for ci-build
+          command: |
+            git config user.email "ci-build@rally-sdk"
+            git config user.name "ci-build"
+      - run:
+          name: Checkout deploy branch and merge onto current branch
+          command: git checkout deploy && git merge --no-ff -m"automated merge to deploy branch" ${CIRCLE_BRANCH}
```

Comment:
We only do this for rally-web-plaform at the request of SRE (because their usual tools like Terraform can't handle Firebase deployment), but I'm fine using the same approach here if you like.

Reply:
An alternative would be to have Circle store artifacts: https://circleci.com/docs/2.0/artifacts/

Reply:
Oh great! I was wondering why we didn't use artifacts and resorted to branch checkin and now I know. I will remove that and just use artifacts. Will post PR update soon.